pkg/antctl/raw/upgrade/apistorage/command.go

```diff
@@ -0,0 +1,292 @@
+		if len(crd.Status.StoredVersions) == 1 && crd.Status.StoredVersions[0] == getCRDStorageVersion(crd) {
+			fmt.Fprintf(writer, "Skip upgrading CRD %q since all stored objects are in the latest version.\n", crdName)
+			continue
+		}
+		crdsToUpgrade = append(crdsToUpgrade, crd)
+	}
+	return crdsToUpgrade, nil
+}
+
+// upgradeCRDObjects upgrades the existing objects of a CRD.
+func upgradeCRDObjects(writer io.Writer, k8sClient client.Client, crd *apiextv1.CustomResourceDefinition) error {
+	objList := &unstructured.UnstructuredList{}
+	objList.SetGroupVersionKind(schema.GroupVersionKind{
+		Group:   crd.Spec.Group,
+		Version: getCRDStorageVersion(crd),
+		Kind:    crd.Spec.Names.ListKind,
+	})
+
+	if err := k8sClient.List(context.TODO(), objList); err != nil {
+		return err
+	}
+
+	if len(objList.Items) == 0 {
+		return nil
+	}
+
+	fmt.Fprintf(writer, "Upgrading %d objects of CRD %q.\n", len(objList.Items), crd.Name)
+	for _, item := range objList.Items {
+		if err := upgradeCRDObject(writer, k8sClient, crd, item); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("error upgrading object %s of CRD %q: %w", k8s.NamespacedName(item.GetNamespace(), item.GetName()), crd.Name, err)
+		}
+	}
+	fmt.Fprintf(writer, "Successfully upgraded %d objects of CRD %q.\n", len(objList.Items), crd.Name)
+
+	return nil
+}
+
+func upgradeCRDObject(writer io.Writer, k8sClient client.Client, crd *apiextv1.CustomResourceDefinition, item unstructured.Unstructured) error {
+	retryOnConflict := false
+
+	return retry.OnError(wait.Backoff{
+		Duration: time.Second,
+		Steps:    5},
+		func(err error) bool {
+			// If the error is that the object is not found, don't retry; retry for other errors.
+			if apierrors.IsNotFound(err) {
+				fmt.Fprintf(writer, "Skip upgrading object %s of CRD %q which is not found.\n", k8s.NamespacedName(item.GetNamespace(), item.GetName()), crd.Name)
+				return false
+			}
+			// If the error is conflict, retry upgrading the object.
+			if apierrors.IsConflict(err) {
+				fmt.Fprintf(writer, "Got conflict error when upgrading object %s of CRD %q, retry upgrading.\n", k8s.NamespacedName(item.GetNamespace(), item.GetName()), crd.Name)
+				retryOnConflict = true
+			}
+			return true
+		},
+		func() error {
+			var objToUpdate *unstructured.Unstructured
+
+			// When retry on a conflict error, get the latest updated object.
+			if retryOnConflict {
+				defer func() {
+					retryOnConflict = false
+				}()
+
+				key := client.ObjectKeyFromObject(&item)
+				objToUpdate = &unstructured.Unstructured{}
+				objToUpdate.SetGroupVersionKind(schema.GroupVersionKind{
+					Group:   crd.Spec.Group,
+					Version: getCRDStorageVersion(crd),
+					Kind:    crd.Spec.Names.Kind,
+				})
+				if err := k8sClient.Get(context.TODO(), key, objToUpdate); err != nil {
+					return err
+				}
+			} else {
+				objToUpdate = &item
+			}
+
+			return k8sClient.Update(context.TODO(), objToUpdate)
```

Comment:
This could be simpler, see how this case is handled in other places: https://github.com/antrea-io/antrea/blob/fea8ddd6078d51be35053710ed5abe90b9a8374f/pkg/controller/networkpolicy/status_controller.go#L375-L382
Just refresh `objToUpdate` when meeting a conflict error.

Reply:
Updated as the example.